### PR TITLE
Fix/new architecture

### PR DESCRIPTION
**Pull request by ignacio-lo-rootstrap**
**Associated tickets**:

- https://github.com/rootstrap/flutter-base/issues/41
- https://github.com/rootstrap/flutter-base/issues/45
- https://github.com/rootstrap/flutter-base/issues/40

**Description**:
 
- New architecture for the base. Previous folders and classes were deleted.

**Overall changes**:

- New folders and classes to follow the new architecture.

**Things to improve**:

- HTTP Client used with Dio is not detecting failures such as no internet.
- Resource class can improve the error handling based on changes when we will have a new HTTP Client.

### CODE REVIEW

**Review by amaury901130 — APPROVED** [review]

**Review by juanRodriguez17 — APPROVED** [review]

File: lib/presentation/themes/theme.dart [hunk 1]
```diff
@@ -139,36 +139,36 @@ abstract class LocalTheme {
         fontFamily: primaryFontBlack,
         fontSize: titleXLFontSize,
         height: titleXLHeight,
-        letterSpacing: -0.07,
+        letterSpacing: -.07,
```
Comment (fabValRoot) [inline lib/presentation/themes/theme.dart:142]: These changes are added to remove the "Double literal shouldn't begin with '.'." warning it generates. Next step is to move them out to remove the "Avoid using magic numbers. Extract them to named constants or variables." I think we should leave them. WDYT

Comment (ignacio-lo-rootstrap) [reply lib/presentation/themes/theme.dart:142]: Got it! I'm gonna look for them in the project and change them

Comment (ignacio-lo-rootstrap) [reply lib/presentation/themes/theme.dart]: Done

**Review by fabValRoot — APPROVED** [review]
Awesome Job Nacho.

**Review by Stukz — APPROVED** [review]

**Pull request merged**